### DISCUSSION TITLE
Interface Event Rule Condition with Device

### DISCUSSION THREAD

**Discussion by PaulR282** [netbox-community/netbox#15284] [Q&A]
Hi,
I have an Event rule that triggers every time an interface is updated. I want to only let it trigger if the device of the interface has a specific custom field. How do I write the condition?
```
{
    "and": [
        {
            "attr": "device.cf.managed",
            "value": "True"
        }
    ]
}
```
I tried this but it doesn't work (also tried with `or` and `true`). The cf is a bool.

**Comment by candlerb** [comment]
Have you tried `"value": True`, without the quotes around True? (That makes it a boolean True, rather than the string "True")

**Reply by PaulR282** [reply]
This does not work because it doesn't validate it as a JSON. 
Edit: It validates with `true` instead of `True` but it still does not seem to work.

**Comment by markkuleinio** [comment]
So your triggering object is an interface, and the interface has an attribute of `device`, right. But since the CF attributes of the device are not part of the data retrieved with the interface, I'm not sure you can match the custom fields of the device here.

Someone feel free to correct me if needed, I haven't used conditions with anything useful yet.

**Reply by PaulR282** [reply]
So basically the same I just figured out. But thanks anyway for the confirmation.

**Comment by PaulR282** [comment]
Update: I dont think this is possible, because the only device data I see in the job data that is being parsed is `id`, `url`, `name`, `display`.

**Reply by multco-troy** [reply]
I've been kicking around how to do this too.  It would be nice to be able to fire off a job when an interface is updated (say the untagged VLAN is changed) but only if the device has a tag, or is active, or?
Firing webhooks off anytime an interface is updated seems like overkill, but maybe that's the way, and then filtering the jobs will be adequate.

**Reply by PaulR282** [reply]
That's my current solution. I trigger the script every time an interface is updated and in the script, I check for the tag.

**Comment by gcarey3** [comment]
I also needed to do an event trigger using a custom field with a boolean type. In my case the data was available in the object. The syntax that worked for me (with version 3.7.1) is this:

```
{
    "and": [
        {
            "attr": "custom_fields.oxidized_enrolled",
            "value": 1
        }
    ]
}
```

**Reply by PaulR282** [reply]
The problem is not on an event rule of a device but on an interface. To my knowledge, you can't access the custom field of the device when triggering an event with the interface.

**Reply by gcarey3** [reply]
Yes, I realize that. I was just pointing out the correct syntax for triggering on a custom field. It's a bit confusing because using the RESTful interface you need to preface the variable with "cf_" but internally to Netbox it looks more python-like.

I think I will have the same challenge when I start automating interface provisioning. I think the only way to trigger on an interface-only attribute and then access the interface would be to point to a script that makes a secondary call to get the interface info.
